core/lib/zksync_core/src/commitment_generator/mod.rs

```diff
@@ -210,11 +214,32 @@ impl CommitmentGenerator {
             }
             state_diffs.sort_unstable_by_key(|rec| (rec.address, rec.key));
 
+            let kzg_settings = load_kzg_settings();
+
+            let blob_commitments = if protocol_version.is_post_1_4_2() {
+                let blob_commitments = header
+                    .pubdata_input
+                    .expect("pubdata_input must be present for post 1.4.2 batches")
+                    .chunks(ZK_SYNC_BYTES_PER_BLOB)
+                    .map(|blob| {
+                        let kzg_info = KzgInfo::new(&kzg_settings, blob.to_vec());
+                        H256(kzg_info.to_blob_commitment())
+                    })
+                    .collect::<Vec<_>>();
+                [
+                    blob_commitments.get(0).copied().unwrap_or_default(),
```

Comment:
Add a comment what happens when pubdata_input is < 128kb
(that the second commitment will be computed as 0s)